platform-metrics/src/test/java/org/hypertrace/core/serviceframework/metrics/PlatformMetricsRegistryTest.java

```diff
@@ -151,9 +154,61 @@ public void testDistributionSummary() {
     assertEquals(1, distribution.count());
     assertEquals(100, distribution.totalAmount());
     assertTrue(
-        Arrays.stream(distribution.takeSnapshot().percentileValues()).map(m -> m.percentile())
-            .collect(
-                Collectors.toList()).containsAll(List.of(0.5, 0.95, 0.99)));
+        Arrays.stream(distribution.takeSnapshot().percentileValues())
+            .map(m -> m.percentile())
+            .collect(Collectors.toList())
+            .containsAll(List.of(0.5, 0.95, 0.99)));
+  }
+
+  @Test
+  public void testCache() throws Exception {
+    initializeCustomRegistry(List.of("testing"));
+    Cache<String, Integer> cache = CacheBuilder.newBuilder().maximumSize(10).recordStats().build();
+
+    PlatformMetricsRegistry.registerCache("my.cache", cache);
+    Callable<Integer> loader =
+        new Callable<Integer>() {
+          @Override
+          public Integer call() throws Exception {
+            return -1;
+          }
+        };
+
+    // Doing some cache activity
+    cache.put("One", 1);
+    cache.put("Two", 2);
+    cache.get("One", loader);
+    cache.get("Two", loader);
+    cache.get("Two", loader);
+    cache.get("Three", loader);
+    cache.get("Failed", loader);
```

Comment:
Not sure why you are calling this `Failed`. It will just load from the loader similar to `Three`

So, the way i understand it, 
- `One` and `Two` were already in the cache. You are getting `One` once and `Two` twice -> which contributes to the hit count of 3
- get on `Three` and `Failed`(which i think you should just rename to may `Four`) goes through the loader and hence are counted as miss
- Cache has 4 entries `One`, `Two`, `Three` and `Failed/Four` -> thats why the cache size is 4

Having a comment explaining the above will simplify things for others who look at the code.

Reply:
The reason for calling Failed is nothing but to actually increase the miss count (similar to Three, nothing special).  
Yes the reason you quote for the cache size being 4 is spot on. I actually included a comment where-in I wrote cache-size = 2 + 2 (misses) = 4.
Maybe I should make it more verbose.

Reply:
Made the comments more verbose by including the meaning of the stats and the way they are calculated.